backend/src/main/java/io/diveni/backend/service/projectmanagementproviders/github/GithubService.java

```diff
@@ -0,0 +1,300 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import jakarta.annotation.PostConstruct;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.server.ResponseStatusException;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import io.diveni.backend.Utils;
+import io.diveni.backend.controller.ErrorMessages;
+import lombok.Getter;
+import lombok.val;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+
+@Service
+public class GithubService implements ProjectManagementProviderOAuth2 {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GithubService.class);
+
+  @Getter private final Map<String, String> accessTokens = new HashMap<>();
+
+  @Getter private final Map<String, String> userNames = new HashMap<>();
+
+  @Getter private final Map<String, String> projectNames = new HashMap<>();
+
+  @Getter private final Map<String, String> selectedProject = new HashMap<>();
+
+  private boolean serviceEnabled = false;
+
+  @PostConstruct
+  public void logConfig() {
+    serviceEnabled = true;
+    LOGGER.info("Github Service: (enabled:" + serviceEnabled + ")");
+  }
+
+  public ResponseEntity<String> executeRequest(
+      String url, HttpMethod method, String token, Object body) {
+    LOGGER.debug("--> executeRequest()");
+    // Create a RestTemplate object
+    RestTemplate restTemplate = new RestTemplate();
+    // Set the headers for the request
+    HttpHeaders headers = new HttpHeaders();
+    headers.setAccept(List.of(MediaType.APPLICATION_JSON));
+    headers.setContentType(MediaType.APPLICATION_JSON);
+    if (!token.equals("")) {
+      headers.add("Authorization", "Bearer " + token);
+    }
+    HttpEntity<Object> request = new HttpEntity<>(body, headers);
+    LOGGER.debug("<-- executeRequest()");
+    return restTemplate.exchange(url, method, request, String.class);
+  }
+
+  public void executeRequestForPatch(String url, HttpMethod method, String token, Object body)
+      throws IOException {
+    LOGGER.debug("--> executeRequest()");
+    CloseableHttpClient httpClient = HttpClients.createDefault();
```

Comment:
It might be a better and safer approach to generate a try-catch block for the httpClient and close it either in a finally block or let it automatically close, using try-with-resources.